src/fragments/lib/graphqlapi/native_common/advanced-workflows/common.mdx

```diff
@@ -90,6 +90,52 @@ import flutter3 from '/src/fragments/lib/graphqlapi/flutter/advanced-workflows/2
 
 <Fragments fragments={{ flutter: flutter3 }} />
 
+<InlineFilter filters={['swift']}>
+
+## Custom Request Inputs
+
+The **GraphQLRequest**'s variables property takes in a JSON object. You can define this as an Encodable type and serialize it to a JSON object for the request.
+
+Define the type that matches the expected GraphQL input type:
+
+```swift
+struct CreateTodoInput: Encodable {
+  let id: String!
+  let name: String!
+  let description: String?
+}
+```
+
+Then create the GraphQLRequest
+```swift
+extension GraphQLRequest {
+    static func createToDo(input: CreateTodoInput) throws -> GraphQLRequest<Todo> {
+      let operationName = "createTodo"
+      let document = """
+      query createTodo($id: ID!) {
+        \(operationName)(id: $id) {
+          id
+          name
+          description
+        }
+      }
+      """
+
+      let data = try JSONEncoder().encode(input)
+      let jsonObject = try JSONSerialization.jsonObject(with: data, options: [])
+
+      return GraphQLRequest<Todo>(
+          document: document,
+          variables: ["input": jsonObject],
+          responseType: Todo.self,
+          decodePath: operationName
+      )
+    }
+}
+```
+
+</InlineFilter>
```

Comment:
```suggestion
```

The `input` object is encoded as a data object and then serialized into the JSON object.

</InlineFilter>
```